From 57f13570023653f85d0920ea64ae3433216c68fc Mon Sep 17 00:00:00 2001
From: Dusty Mabe <dusty@dustymabe.com>
Date: Mon, 27 Nov 2023 16:14:43 -0500
Subject: [PATCH 1/3] osbuild/util/ostree: convert cli to return the completed
 process object

And also set stdout=subprocess.PIPE. This will allow for callers to
parse and use the output of the command, but has the side effect of
meaning less gets printed to the screen during run.

Co-authored-by: Luke Yang <luyang@redhat.com>
---
 osbuild/util/ostree.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/osbuild/util/ostree.py b/osbuild/util/ostree.py
index 43dd99c5..09c648fb 100644
--- a/osbuild/util/ostree.py
+++ b/osbuild/util/ostree.py
@@ -199,11 +199,11 @@ def cli(*args, _input=None, **kwargs):
     """Thin wrapper for running the ostree CLI"""
     args = list(args) + [f'--{k}={v}' for k, v in kwargs.items()]
     print("ostree " + " ".join(args), file=sys.stderr)
-    subprocess.run(["ostree"] + args,
-                   encoding="utf8",
-                   stdout=sys.stderr,
-                   input=_input,
-                   check=True)
+    return subprocess.run(["ostree"] + args,
+                          encoding="utf8",
+                          stdout=subprocess.PIPE,
+                          input=_input,
+                          check=True)
 
 
 def parse_input_commits(commits):
-- 
2.41.0

